code/game/atoms_fluids.dm

```diff
@@ -4,7 +4,7 @@
 /atom/proc/fluid_act(var/datum/reagents/fluids)
 	SHOULD_CALL_PARENT(TRUE)
 	// TODO: fix food open container behavior jesus christ
-	if(reagents && reagents != fluids && fluids?.total_volume >= FLUID_SHALLOW && ATOM_IS_OPEN_CONTAINER(src) && !istype(src, /obj/item/chems/food))
+	if(reagents && reagents != fluids && fluids?.total_volume >= FLUID_SHALLOW && !is_watertight() && !istype(src, /obj/item/chems/food))
```

Comment:
when this is pulled into dev, do we want to set `watertight = TRUE` on lanterns? or would we rather just add a toggleable cap?